src/admin_ws.rs

```diff
@@ -0,0 +1,203 @@
+use holochain_client::{AdminWebsocket, ConductorApiError, ConductorApiResult};
+use std::{future::Future, sync::Arc};
+use tokio::sync::RwLock;
+use url2::Url2;
+
+use crate::{HcHttpGatewayError, HcHttpGatewayResult};
+
+const ADMIN_WS_CONNECTION_MAX_RETRIES: usize = 1;
+
+/// A wrapper around AdminWebsocket that automatically handles reconnection
+/// when the connection is lost due to network issues or other failures.
+#[derive(Debug, Clone)]
+pub struct HcHttpGwAdminWebsocket {
+    /// The WebSocket URL to connect to
+    url: Url2,
+    /// The handle to the AdminWebsocket connection
+    connection_handle: Arc<RwLock<Option<AdminWebsocket>>>,
```

Comment:
Why is this an `Option<AdminWebsocket>`? It looks like it needs to make a successful connection within the `connect` constructor, so if the struct is instantiated then we should always have an `AdminWebsocket`